spec/models/xml_source_processor_spec.rb

```diff
@@ -1,76 +1,78 @@
-      </page>
+SOURCE_TEXT = 'With an [[Old Subject|old subject]] and a short [[Old Subject]]. With a [[New Text Link|new links]] and a [[New Short Text Link]]'
+EXPECTED_XML = <<~EOF
+  <?xml version='1.0' encoding='UTF-8'?>
+        <page>
+          <p>With an <link link_id='1' target_id='1' target_title='Old Subject'>old subject</link> and a short <link link_id='2' target_id='1' target_title='Old Subject'>Old Subject</link>. With a <link link_id='3' target_id='2' target_title='New Text Link'>new links</link> and a <link link_id='4' target_id='3' target_title='New Short Text Link'>New Short Text Link</link></p>
+        </page>
 EOF
 
-EXPECTED_XML_DISABLED = <<EOF
-<?xml version='1.0' encoding='UTF-8'?>    
-      <page>
-        <p>With an [[Old Subject|old subject]] and a short [[Old Subject]]. With a [[New Text Link|new links]] and a [[New Short Text Link]]</p>
-      </page>
+EXPECTED_XML_DISABLED = <<~EOF
+  <?xml version='1.0' encoding='UTF-8'?>
+        <page>
+          <p>With an [[Old Subject|old subject]] and a short [[Old Subject]]. With a [[New Text Link|new links]] and a [[New Short Text Link]]</p>
+        </page>
 EOF
 
 SOURCE_TEXT_ILLEGAL_CHARS = "\fWith a lo\vad of illegal \u000C charac\u0014ters and a tab\t"
-EXPECTED_XML_ILLEGAL_CHARS = <<EOF
-<?xml version='1.0' encoding='UTF-8'?>    
-      <page>
-        <p> With a lo ad of illegal   charac ters and a tab\t</p>
-      </page>
+EXPECTED_XML_ILLEGAL_CHARS = <<~EOF
+  <?xml version='1.0' encoding='UTF-8'?>
+        <page>
+          <p> With a lo ad of illegal   charac ters and a tab\t</p>
+        </page>
 EOF
 
 RSpec.describe XmlSourceProcessor, type: :model do
   describe '#wiki_to_xml' do
-  
-  before :each do
-    DatabaseCleaner.clean_with(:truncation)
-  end
+    before :each do
+      DatabaseCleaner.clean_with(:truncation)
+    end
+
+    let(:collection) { build_stubbed(:collection) }
+    let(:work)      { build_stubbed(:work, collection:) }
+    let(:page)      { build_stubbed(:page, work:, source_text: SOURCE_TEXT) }
+    let(:old_link)  { build_stubbed(:article, title: 'Old Subject', collection:) }
 
-  let(:collection){ build_stubbed(:collection ) }
-  let(:work)      { build_stubbed(:work, collection: collection) }
-  let(:page)      { build_stubbed(:page, work: work, source_text: SOURCE_TEXT)}
-  let(:old_link)  { build_stubbed(:article, title: 'Old Subject', collection: collection ) }
-    
     context 'subject linking not disabled (default)' do
       it 'builds the xml document' do
         expect(work.collection).to eq(collection)
         xml = page.wiki_to_xml(page, Page::TEXT_TYPE::TRANSCRIPTION)
         expect(Article.all.count).to eq(3)
         expect(PageArticleLink.all.count).to eq(4)
-        expect(xml).to eq(EXPECTED_XML)
+        expect(Nokogiri::XML(xml).to_s).to eq(Nokogiri::XML(EXPECTED_XML).to_s)
```

Comment:
For some reason eq() matching is failing due to inconsistent tabs/spaces, but contents are literally the same. Doing this ensures we compare the contents.